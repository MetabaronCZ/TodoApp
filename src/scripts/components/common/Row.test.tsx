import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Row } from 'components/common/Row';
import { withMockedProviders } from 'test/component';

describe('components/common/Row', () => {
  it('should render correctly', () => {
    const tree = render(withMockedProviders(<Row>CONTENT</Row>));
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByText('CONTENT')).toBeInTheDocument();
  });

  it('should render props', () => {
    const tree = render(withMockedProviders(<Row size={4} />));
    expect(tree.container).toMatchSnapshot();
    expect(tree.container.firstChild).toHaveStyle({ gap: '40px' });
  });

  it('should keep minimum prop size', () => {
    const tree = render(withMockedProviders(<Row size={-2} />));
    expect(tree.container.firstChild).toHaveStyle({ gap: '0px' });
  });
});
