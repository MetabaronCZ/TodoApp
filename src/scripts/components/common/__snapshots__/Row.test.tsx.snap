// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/common/Row > should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 20px;
}

.c0 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div
    class="c0"
  >
    CONTENT
  </div>
</div>
`;

exports[`components/common/Row > should render props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 40px;
}

.c0 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div
    class="c0"
  />
</div>
`;
